src/nvim/decoration.c

```diff
@@ -218,6 +219,38 @@ bool decor_redraw_line(buf_T *buf, int row, DecorState *state)
   return true;  // TODO(bfredl): be more precise
 }
 
+static void hlrange_activate(HlRange range, DecorState *state)
+{
+  // Get size before preparing the push, to have the number of elements
+  size_t s = kv_size(state->active);
+
+  HlRange *end_pos = kv_pushp(state->active);
+
+  size_t dest_index = 0;
+
+  // Determine insertion dest_index
+  while (dest_index < s) {
+    HlRange item = kv_A(state->active, dest_index);
+    if (item.priority < range.priority) {
+      break;
+    }
+
+    dest_index++;
+  }
+
+  if (dest_index == s) {
```

Comment:
Branch not needed

Reply:
I intially thought this was an optimization, but in the end we don't gain that much